cmd/cloud_run.go

```diff
@@ -38,12 +79,87 @@ Use the "k6 cloud login" command to authenticate.`,
 			"the k6 cloud run command expects a single argument consisting in either a path to a script or "+
 				"archive file, or the \"-\" symbol indicating the script or archive should be read from stdin",
 		),
-		PreRunE: deprecatedCloudCmd.preRun,
-		RunE:    deprecatedCloudCmd.run,
+		PreRunE: cloudRunCmd.preRun,
+		RunE:    cloudRunCmd.run,
+	}
+
+	thisCmd.Flags().SortFlags = false
+	thisCmd.Flags().AddFlagSet(cloudRunCmd.flagSet())
+	thisCmd.Flags().AddFlagSet(cloudCmd.flagSet())
+
+	return thisCmd
+}
+
+func (c *cmdCloudRun) preRun(cmd *cobra.Command, args []string) error {
+	if c.localExecution {
+		return nil
+	}
+
+	if c.linger {
+		return fmt.Errorf("the --linger flag can only be used in conjunction with the --local-execution flag")
+	}
+
+	if c.noUsageReport {
+		return fmt.Errorf("the --no-usage-report can only be used in conjunction with the --local-execution flag")
+	}
+
+	return c.deprecatedCloudCmd.preRun(cmd, args)
+}
+
+func (c *cmdCloudRun) run(cmd *cobra.Command, args []string) error {
+	if c.localExecution {
+		if cmd.Flags().Changed("exit-on-running") {
+			return fmt.Errorf("the --local-execution flag is not compatible with the --exit-on-running flag")
+		}
+
+		if cmd.Flags().Changed("show-logs") {
+			return fmt.Errorf("the --local-execution flag is not compatible with the --show-logs flag")
+		}
+
+		return c.runCmd.run(cmd, args)
+	}
+
+	// When executing in the cloud, we enforce the usage report to be deactivated.
+	c.noUsageReport = true
```

Comment:
I guess in the future, we'll most likely change this for the 2nd iteration of https://github.com/grafana/k6-cloud/issues/2613 cc/ @codebien

Reply:
No, this isn't a control we enable/disable here, we do it directly using the env var. Do we need this change?

Reply:
I'm sorry I don't understand. Should we keep the no usage report option, and should we force it to be set to true in this context?

Reply:
We should have the same concept as we have right now with `k6 run -o cloud script.js`. In that case, the report is enabled.

If the report is enabled or not in the Cloud is not controlled by k6 but it is injected from the Cloud at runtime.

Reply:
So my understanding is that the actionables step for us here would be:
1. keep the option to support `k6 cloud run --local-execution --no-usage-report`
2. remove forcing it to true, as in this snippet, in the case of `k6 cloud run`, as it won't be useful anyway.

Is that correct?

Reply:
>as it won't be useful anyway.

No, the risk is quite the opposite. It has an impact and it might generate unexpected conditions.

But let me restart because maybe I'm missing some context.

Why did we write this line change (`c.noUsageReport = true`)? What are the problem and the solution proposal?

Reply:
We added the `--linger` and `--no-usage-report` options to `k6 cloud run`, which do something when the `--local-execution` option is used. We want to reproduce the same behavior as if people used the `k6 run` command with the `-o cloud` option.

This specific line of code has been added to simulate that the `--no-usage-report` option we've added is always passed (set to true) if the user does not use the `--local-execution`; meaning they're using the `k6 cloud run` command and execute their test in the cloud, as opposed to locally.

Reply:
I have changed the behavior of the command according to our internal discussions:
* `k6 cloud run --no-usage-report` should ignore the flag, and make sure we don't send the report, simulating the behavior.
* `k6 cloud run --local-execution --no-usage-report`  should take the flag in consideration, and act accordingly.